go-controller/pkg/ovn/controller/apbroute/external_controller_test.go

```diff
@@ -0,0 +1,119 @@
+package apbroute
+
+import (
+	"fmt"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+var (
+	// Default comparable options to bypass comparing the mux object and define a less function to sort the slices.
+	cmpOpts = []cmp.Option{
+		cmpopts.IgnoreFields(gatewayInfo{}, "mux"),
+		cmpopts.SortSlices(func(x, y interface{}) bool {
+			return fmt.Sprint(x) < fmt.Sprint(y)
+		})}
```

Comment:
Two comments regarding this:

- Are you reusing the same options to compare more than one different type?
- Is this leading to using the matcher `BeComparableTo` when `BeEqual` should be enough?

For example, I don't see what is the use of this `cmpOpts` when you use it to compare a single `namespaceInfo`? Since its noty a slice and it doesn't have the `mux` field.

Reply:
Ok, I didn't expect it to be recursive